### PR TITLE
Fixes #266: Introduce layout commands for sizing and aligning elements

### LINKED ISSUE #266: Layout Commands for simplifying manual layouting
Implement layout commands that help the user with common layout tasks like:
-  same size as the first selected element
-align to left/right of the selected element
- align to top/bottom of the selected element 

(right-click menu and keyboard shortcut on multi-select)

### PR DESCRIPTION
**Pull request by martin-fleck-at**
Action and command to resize given or selected elements to:
- min/max/avg/first/last width or height

Action and command to align given or selected elements to:
- left/center/right for all/first/last selected element(s)
- top/middle/bottom for all/first/last selected element(s)

Use keybindings within Sprotty for some actions
Register actions in the 'Diagram' menu for Theia

### Align
align

### Resize
resize

### CODE REVIEW

File: client/packages/sprotty-client/src/features/layout/layout-commands.ts [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+         */
+        public readonly alignment: Alignment = Alignment.Left,
+        /**
+         * Function to selected elements that are considered during alignment calculation, see Select.* for potential functions.
+         */
+        public readonly selectionFunction: (elements: SelectableBoundsAware[]) => SelectableBoundsAware[] = Select.all) {
+    }
+}
+
+abstract class LayoutElementsCommand extends Command {
```
Comment (tortmayr) [inline client/packages/sprotty-client/src/features/layout/layout-commands.ts]: According to sprotty guidelines, commands should be annotated with `@injectable`

File: client/packages/sprotty-client/src/features/layout/layout-commands.ts [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+        this.actionDispatcher.dispatch(action);
+    }
+
+    dispatchActions(actions: Action[]) {
+        this.actionDispatcher.dispatchAll(actions);
+    }
+}
+
+export class ResizeElementsCommand extends LayoutElementsCommand {
+    static readonly KIND = 'layout:resize';
```
Comment (tortmayr) [inline client/packages/sprotty-client/src/features/layout/layout-commands.ts]: According to sprotty guidelines, commands should be annotated with `@injectable`

File: client/packages/sprotty-client/src/features/layout/layout-commands.ts [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+        return context.root;
+    }
+
+    redo(context: CommandExecutionContext): CommandResult {
+        // we dispatch another action which can be redone, so no explicit implementation necessary
+        return context.root;
+    }
+}
+
+export class AlignElementsCommand extends LayoutElementsCommand {
```
Comment (tortmayr) [inline client/packages/sprotty-client/src/features/layout/layout-commands.ts]: According to sprotty guidelines, commands should be annotated with `@injectable`

File: client/packages/sprotty-client/src/features/layout/layout-commands.ts (hunk 1 above)
Comment (martin-fleck-at) [reply client/packages/sprotty-client/src/features/layout/layout-commands.ts]: Done

File: client/packages/sprotty-client/src/features/layout/layout-commands.ts (hunk 2 above)
Comment (martin-fleck-at) [reply client/packages/sprotty-client/src/features/layout/layout-commands.ts]: Done

File: client/packages/sprotty-client/src/features/layout/layout-commands.ts (hunk 3 above)
Comment (martin-fleck-at) [reply client/packages/sprotty-client/src/features/layout/layout-commands.ts]: Done

**Comment by martin-fleck-at** [comment]
Hi Tobi, I addressed your feedback regarding the sprotty guidelines in a follow-up commit and created a new issue for the consideration of the overlap setting: https://github.com/eclipsesource/graphical-lsp/issues/271

**Pull request merged**